src/cursor/cur_version.c

```diff
@@ -620,19 +611,30 @@ __wt_curversion_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner
     WT_CURSOR *cursor;
     WT_CURSOR_VERSION *version_cursor;
     WT_DECL_RET;
+    WT_TXN_GLOBAL *txn_global;
+    wt_timestamp_t pinned_ts;
     /* The file cursor is read only. */
     const char *file_cursor_cfg[] = {
       WT_CONFIG_BASE(session, WT_SESSION_open_cursor), "read_only=true", NULL};
     char *version_cursor_value_format;
     size_t format_len;
 
+    txn_global = &S2C(session)->txn_global;
     *cursorp = NULL;
     WT_RET(__wt_calloc_one(session, &version_cursor));
     cursor = (WT_CURSOR *)version_cursor;
     *cursor = iface;
     cursor->session = (WT_SESSION *)session;
     version_cursor_value_format = NULL;
 
+    /* Freeze pinned timestamp when we open the first version cursor. */
+    if (__wt_atomic_add32(&S2C(session)->version_cursor_count, 1) == 1) {
```

Comment:
The version count is incremented and later the pinned_timestamp function is called and inside that pinned timestamp we return version_cursor_pinned_timestamp that didn't set any value. So this leads to a scenario the pinned timestamp will be zero.

Reply:
good catch

Reply:
Hmm, instead of another flag variable, how about checking the condition for == 0 before incrementing it, so we don't have to pass the flags everywhere?

Reply:
Yes, good idea.